config/proposer/loader.go

```diff
@@ -0,0 +1,281 @@
+package proposer
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v5/cmd/validator/flags"
+	"github.com/prysmaticlabs/prysm/v5/config"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
+	validatorService "github.com/prysmaticlabs/prysm/v5/config/validator/service"
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/validator"
+	validatorpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1/validator-client"
+	"github.com/prysmaticlabs/prysm/v5/validator/db/iface"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+type SettingsType int
+
+const (
+	none SettingsType = iota
+	defaultFlag
+	fileFlag
+	urlFlag
+	onlyDB
+)
+
+type SettingsLoader struct {
+	loadMethods []SettingsType
+	existsInDB  bool
+	db          iface.ValidatorDB
+	options     *flagOptions
+}
+
+type flagOptions struct {
+	builderConfig *validatorService.BuilderConfig
+	gasLimit      validator.Uint64
+	hasGasLimit   bool
+}
+
+// SettingsLoaderOption sets additional options that affect the proposer settings
+type SettingsLoaderOption func(cliCtx *cli.Context, psl *SettingsLoader) error
```

Comment:
No need to export.

Reply:
I also need to export using the withbuilder config

Reply:
Hum everything seems to work fine without exporting on my side.